src/SonarScanner.MSBuild.Tasks/ProjectConfig.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * SonarScanner for MSBuild
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Xml.Serialization;
+
+namespace SonarScanner.MSBuild.Common
+{
+    /// <summary>
+    /// Data class to describe a single project configuration for our analyzers.
+    /// </summary>
+    /// <remarks>The class is XML-serializable</remarks>
+    [XmlRoot(ElementName = "SonarProjectConfig", Namespace = "http://www.sonarsource.com/msbuild/analyzer/2021/1")]
+    public class ProjectConfig
+    {
+        /// <summary>
+        /// Full path to the SonarQubeAnalysisConfig.xml file.
+        /// </summary>
+        public string AnalysisConfigPath { get; set; }
+
+        /// <summary>
+        /// The full name and path of the project file.
+        /// </summary>
+        public string ProjectPath { get; set; }
+
+        /// <summary>
+        ///The full name and path of the text file containing all files to analyze.
+        /// </summary>
+        public string FilesToAnalyzePath { get; set; }
+
+        /// <summary>
+        /// Root of the project-specific output directory. Analyzer should write protobuf and other files there.
+        /// </summary>
+        public string OutPath { get; set; }
+
+        /// <summary>
+        /// The kind of the project.
+        /// </summary>
+        public ProjectType ProjectType { get; set; }
+
+        /// <summary>
+        /// MSBuild target framework for the current build.
+        /// </summary>
+        public string TargetFramework { get; set; }
+
+        /// <summary>
+        /// Saves the project configuration to the specified file as XML.
+        /// </summary>
```

Comment:
maybe mention a remark that both `Save` and `Load` throw exceptions which need to be handled by the caller?

Reply:
`ProjectInfo.cs` doesn't have it so I think we don't need it. Possible exception is expected in any method.